Improvements

alexaandrov opened the pull request.

alexaandrov committed:
Added tabs navigation

alexaandrov committed:
Fixed start linewise visual mode command behavior

Fixed incorrect line selection behavior (shift + v) when the number of lines is 1 or if the line is the last

lubokkanev reviewed (approved):
Looks good!

lubokkanev reviewed (changes requested):
I think it's good the way it is, but if you come up with a better alternative, it'd be great.

lubokkanev commented on src/shortcuts/mode-switchers.ahk:
```diff
@@ -28,6 +28,5 @@ v::
 +v::
     turnVisualModeOn()
-    Send, {Home}{Home}+{Down}
+    Send, {End}+{Home}
```
I think the most popular usecase for (shift + v) is selecting the current line and the moving down. `{End}+{Home}` breaks that, because trying to select the lines below unselects the current line. 

`{Home}+{Down}` has the same problem when going up, but I think that's less often used, so it's a viable tradeoff. If there's a way to work in both directions, that'd be great.

lubokkanev commented on src/shortcuts/mode-switchers.ahk:
```diff
@@ -28,6 +28,5 @@ v::
 
 +v::
     turnVisualModeOn()
-    Send, {Home}{Home}+{Down}
```
In many editors, like Sublime and IntelliJ for example, pressing Home a single time doesn't move you to the beginning of the line, but to the end of the initial whitespaces. A double Home fixes that. On editors that don't behave like that, double Home doesn't bring any problems either, so it's a viable solution. 

I don't think there are problems with `{Home}{Home}+{Down}` on the first and last lines, except the ones specified in the comment below.

alexaandrov committed:
Improvements
    - Added mouse control (ctrl + [) - (jasd, n, m)
    - Added abbility to disable and enable vim mode
    - Added GUI label of current mode
    - Refactoring modes
    - Added capslock arrows to navigate win os windows
